pkg/frontend/validate.go

```diff
@@ -64,7 +65,7 @@ func (f *frontend) validateOpenShiftUniqueKey(ctx context.Context, doc *api.Open
 	if docs.Count != 0 {
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeDuplicateResourceGroup, "", "The provided resource group '%s' already contains a cluster.", doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID)
 	}
-	return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal server error.")
+	return api.NewInternalServerError(ctx.Value(middleware.ContextKeyCorrelationData).(*api.CorrelationData))
```

Comment:
```suggestion
	return errors.New("failed to resolve unique key error")
```
should do ^ instead of building a CloudError here.